src/api/goods.ts

```
@@ -4,13 +4,27 @@ import { Good } from '../types/Good';
 const API_URL = `https://mate-academy.github.io/react_dynamic-list-of-goods/goods.json`;
 
 export function getAll(): Promise<Good[]> {
-  return fetch(API_URL).then(response => response.json());
+  return fetch(API_URL)
+    .then(response => {
+      if (!response.ok) {
+        return Promise.reject(new Error(`HTTP error: ${response.status}`));
+      }
+
+      return response.json();
+    })
+    .catch(error => {
+      throw error('Failed to fetch goods:', error);
```

This `throw` statement is still not quite right. The `throw` keyword expects a single value to be thrown (like an `Error` object), but the current syntax `error(...)` attempts to call the `error` object as a function, which will result in a `TypeError`.

To fix this and correctly propagate the error, you can simply re-throw the original error that was caught:

```javascript
.catch(error => {
  // Re-throw the original error
  throw error;
});
```

This is a critical part of fulfilling checklist item #3 regarding error handling.